ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java

```diff
@@ -0,0 +1,137 @@
+package gov.cms.ab2d.bfd.client;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.OffsetDateTime;
+
+import org.hl7.fhir.instance.model.api.IBaseBundle;
+import org.junit.jupiter.api.Test;
+import org.springframework.core.env.Environment;
+
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import gov.cms.ab2d.fhir.FhirVersion;
+
+class BFDSearchImplTest {
+
+  private static int patientId = 1;
+  private static OffsetDateTime since = OffsetDateTime.now();
+  private static OffsetDateTime until = OffsetDateTime.now();
+  private static int pageSize = 10;
+  private static String bulkJobId = "bulkJobId";
+  private static FhirVersion version = FhirVersion.R4;
+  private static String contractNum = "contractNum";
+  private static String filePath = "src/test/resources/bb-test-data/eob/20010000001115.json";
+  private static String activeProfile = "test";
+  private static String bfdUrl = "http://localhost:8080";
+
+  HttpClient httpClient200() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+    when(response.getEntity()).thenReturn(mock(org.apache.http.HttpEntity.class));
+    String fileContent = Files.readString(Paths.get(filePath));
+    when(response.getEntity().getContent()).thenReturn(new ByteArrayInputStream(fileContent.getBytes()));
+    return httpClient;
+  }
```

Comment:
This took me FOREVER to figure out. I hope its okay 😵

Reply:
I am beyond impressed how you found a way to test this , woww!

Reply:
This is what I'm planning on demoing